### PR TITLE
Add a Parinfer User autocommand event

### PR DESCRIPTION
**Pull request by gpanders**

### CODE REVIEW

**Comment by p00f** [comment]
Thanks, this only works when you `:ParinferOn` manually - the filetype autocommand uses parinfer.enter_buffer, not parinfer#enable

**Comment by p00f** [comment]
also ParinferToggle only works if you already have used ParinferOn or ParinferOff - i think you need to use `get()` instead of using the parinfer_enabled variable directly

**Comment by p00f** [comment]
also
```
[I] ganymede ~/.loc/sha/nvi/sit/pac/paq/sta/nvim-parinfer (master % =)> gh pr checkout 17
fatal: cannot set up tracking information; starting point 'origin/parinfer-user-event' is not a branch
failed to run git: exit status 128
[I] ganymede ~/.loc/sha/nvi/sit/pac/paq/sta/nvim-parinfer (master +% =) [1]>
```

**Comment by gpanders** [comment]
> also ParinferToggle only works if you already have used ParinferOn or ParinferOff - i think you need to use `get()` instead of using the parinfer_enabled variable directly

Should be fixed now.

File: autoload/parinfer.vim [hunk 1]
```diff
@@ -52,6 +53,8 @@ function! parinfer#init() abort
 
     lua parinfer.enter_buffer()
```
Comment (p00f) [inline autoload/parinfer.vim:55]: @gpanders  also need a `let b:parinfer_enabled = 1` before the `doautocmd`, then it works

**Pull request merged**